### PR TITLE
DROID-2853 All content | Enhancement | Menu, cache, empty states

### PR DESCRIPTION
**Pull request by konstantiniiv**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: data/src/main/java/com/anytypeio/anytype/data/auth/repo/UserSettingsCache.kt [hunk 1]
```diff
@@ -35,4 +35,7 @@ interface UserSettingsCache {
     suspend fun getWidgetSession() : WidgetSession
     suspend fun saveWidgetSession(session: WidgetSession)
     suspend fun clear()
+
+    suspend fun getAllContentSort(space: SpaceId): String
```
Comment (uburoiubu) [inline data/src/main/java/com/anytypeio/anytype/data/auth/repo/UserSettingsCache.kt:39]: `suspend fun getAllContentSort(space: SpaceId): Id`

File: persistence/src/main/proto/preferences.proto [hunk 2]
```diff
@@ -13,9 +13,14 @@ message SpacePreference {
...
   optional GlobalSearchHistoryProto globalSearchHistory = 5;
+  optional AllContentProto allContent = 6;
 }
 
 message GlobalSearchHistoryProto {
   optional string lastSearchQuery = 1;
   optional string lastSearchRelatedObjectId = 2;
+}
+
+message AllContentProto {
```
Comment (uburoiubu) [inline persistence/src/main/proto/preferences.proto:24]: AllContentSettings

File: persistence/src/main/proto/preferences.proto [hunk 3]
```diff
@@ -13,9 +13,14 @@ message SpacePreference {
   repeated string pinnedObjectTypeIds = 2;
   optional string lastOpenedObject = 3;
   optional GlobalSearchHistoryProto globalSearchHistory = 5;
```
Comment (uburoiubu) [inline persistence/src/main/proto/preferences.proto:15]: VersionHistorySettings

File: presentation/src/main/java/com/anytypeio/anytype/presentation/home/HomeScreenViewModel.kt [hunk 4]
```diff
@@ -410,6 +406,18 @@ class HomeScreenViewModel(
         }
     }
 
+    private fun getListOfWidgets(
```
Comment (uburoiubu) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/home/HomeScreenViewModel.kt:409]: buildListOfWidgets

**Review by uburoiubu — APPROVED** [review]

File: data/src/main/java/com/anytypeio/anytype/data/auth/repo/UserSettingsCache.kt (hunk 1 above)
Comment (konstantiniiv) [reply data/src/main/java/com/anytypeio/anytype/data/auth/repo/UserSettingsCache.kt:39]: done

File: persistence/src/main/proto/preferences.proto (hunk 2 above)
Comment (konstantiniiv) [reply persistence/src/main/proto/preferences.proto:24]: done

File: persistence/src/main/proto/preferences.proto (hunk 3 above)
Comment (konstantiniiv) [reply persistence/src/main/proto/preferences.proto:15]: it's a different feature actually

File: presentation/src/main/java/com/anytypeio/anytype/presentation/home/HomeScreenViewModel.kt (hunk 4 above)
Comment (konstantiniiv) [reply presentation/src/main/java/com/anytypeio/anytype/presentation/home/HomeScreenViewModel.kt:409]: done

**Pull request merged**